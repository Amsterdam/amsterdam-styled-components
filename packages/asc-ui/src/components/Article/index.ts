import ArticleContentStyle from './ArticleContent/ArticleContentStyle'
import ArticleHeaderStyle from './ArticleHeader/ArticleHeaderStyle'
import ArticleSidebarStyle from './ArticleSidebar/ArticleSidebarStyle'
import ArticleMetaListStyle from './ArticleMetaList/ArticleMetaListStyle'

export { default } from './Article'
export { default as ArticleContent } from './ArticleContent'
export { default as ArticleSidebar } from './ArticleSidebar'
export { default as ArticleMetaList } from './ArticleMetaList'
export { default as ArticleHeader } from './ArticleHeader'

export const ArticleStyles = {
  ArticleContentStyle,
  ArticleMetaListStyle,
  ArticleSidebarStyle,
  ArticleHeaderStyle,
}
