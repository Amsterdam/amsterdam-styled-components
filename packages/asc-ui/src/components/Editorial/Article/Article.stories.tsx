import React from 'react'
import articleJSON from './article_feed_example.json'
import {
  Column,
  CustomHTMLBlock,
  Row,
  Heading,
  List,
  ListItem,
  Article,
  EditorialBody,
  Paragraph,
} from '../../..'

/* Editorial components */
import EditorialContent from '../EditorialContent'
import EditorialHeader from '../EditorialHeader'
import EditorialMetaList from '../EditorialMetaList'
import EditorialSidebar from '../EditorialSidebar/EditorialSidebar'
import Link from '../../Link/Link'

export default {
  title: 'Dataportaal/Experimental/Composed/Editorial/Article',

  decorators: [
    (storyFn: () => React.ReactNode) => (
      <div style={{ padding: '40px 10px' }}>{storyFn()}</div>
    ),
  ],
}

export const WithImage = () => (
  <Article image={articleJSON.image}>
    <Row>
      <EditorialContent>
        <Column
          wrap
          span={{ small: 1, medium: 2, big: 5, large: 11, xLarge: 11 }}
          push={{ small: 0, medium: 0, big: 1, large: 1, xLarge: 1 }}
        >
          <Column span={{ small: 1, medium: 2, big: 4, large: 7, xLarge: 7 }}>
            <EditorialBody>
              <EditorialHeader title={articleJSON.title}>
                <EditorialMetaList
                  dateTime={articleJSON.publicationDate}
                  dateFormatted={articleJSON.publicationDateFormatted}
                  fields={[{ id: 1, label: 'Redactie' }]}
                />
              </EditorialHeader>
              <Paragraph strong>{articleJSON.fields.intro}</Paragraph>
              <CustomHTMLBlock body={articleJSON.body} />
            </EditorialBody>
          </Column>
          <Column
            span={{ small: 1, medium: 2, big: 2, large: 3, xLarge: 3 }}
            push={{ small: 0, medium: 0, big: 1, large: 1, xLarge: 1 }}
          >
            <EditorialSidebar>
              <Heading as="h2">Links</Heading>
              <List>
                <ListItem>
                  <Link href="/">I am a ListLink!</Link>
                </ListItem>
                <ListItem>
                  <Link href="/">Another one</Link>
                </ListItem>
                <ListItem>
                  <Link href="/">A loooooong, very, very, very long link</Link>
                </ListItem>
              </List>
            </EditorialSidebar>
          </Column>
        </Column>
      </EditorialContent>
    </Row>
  </Article>
)

WithImage.story = {
  name: 'with image',
}

export const WithoutImage = () => (
  <Article>
    <Row>
      <EditorialContent>
        <Column
          wrap
          span={{ small: 1, medium: 2, big: 5, large: 11, xLarge: 11 }}
          push={{ small: 0, medium: 0, big: 1, large: 1, xLarge: 1 }}
        >
          <Column span={{ small: 1, medium: 2, big: 4, large: 7, xLarge: 7 }}>
            <EditorialBody>
              <EditorialHeader title={articleJSON.title}>
                <EditorialMetaList
                  dateTime={articleJSON.publicationDate}
                  dateFormatted={articleJSON.publicationDateFormatted}
                  fields={[{ id: 1, label: 'Redactie' }]}
                />
              </EditorialHeader>
              <Paragraph strong>{articleJSON.fields.intro}</Paragraph>
              <CustomHTMLBlock body={articleJSON.body} />
            </EditorialBody>
          </Column>
          <Column
            span={{ small: 1, medium: 2, big: 2, large: 3, xLarge: 3 }}
            push={{ small: 0, medium: 0, big: 1, large: 1, xLarge: 1 }}
          >
            <EditorialSidebar>
              <Heading as="h2">Downloads</Heading>
              <List>
                <ListItem>
                  <Link href="/">File</Link>
                </ListItem>
                <ListItem>
                  <Link href="/">File</Link>
                </ListItem>
                <ListItem>
                  <Link href="/">File</Link>
                </ListItem>
              </List>
            </EditorialSidebar>
          </Column>
        </Column>
      </EditorialContent>
    </Row>
  </Article>
)

WithoutImage.story = {
  name: 'without image',
}
