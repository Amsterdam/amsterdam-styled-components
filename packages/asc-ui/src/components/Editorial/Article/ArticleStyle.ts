import styled, { css } from 'styled-components'
import { stripUnit } from 'polished'
import { breakpoint } from '../../../utils'
import EditorialBodyStyle from '../EditorialBody/EditorialBodyStyle'
import { EDITORIAL_PADDING_TOP } from '../EditorialPost/EditorialPostStyle'
import { getTypographyValueFromProperty } from '../../../utils/themeUtils'
import {
  EDITORIAL_META_LINE_HEIGHT,
  EDITORIAL_META_MARGIN_TOP,
} from '../EditorialMetaList/EditorialMetaListStyle'
import EditorialPost, {
  Props as EditorialPostProps,
} from '../EditorialPost/EditorialPost'

export default styled(EditorialPost)<EditorialPostProps>`
  ${EditorialBodyStyle} {
    ${({ image }) =>
      !!image &&
      css`
        @media screen and ${breakpoint('min-width', 'tabletM')} {
          padding: ${EDITORIAL_PADDING_TOP}px 24px;
        }
      `}

    /* 
    Here we calculate how much the EditorialBody needs to shift up by getting existing css values from:
    - The H1 line-height
    - the padding of the EditorialBody
    - the EditorialMeta margin-top and line-height
    */
    @media screen and ${breakpoint('min-width', 'tabletM')} {
      margin-top: ${({ image, theme }) =>
        image
          ? `${
              (EDITORIAL_PADDING_TOP * 2 +
                EDITORIAL_META_LINE_HEIGHT +
                EDITORIAL_META_MARGIN_TOP +
                Number(
                  stripUnit(
                    getTypographyValueFromProperty(
                      'h1',
                      'lineHeight',
                      'tabletS',
                    )({
                      theme,
                    }),
                  ),
                )) *
              -1
            }px`
          : `0`};
    }
  }
`
