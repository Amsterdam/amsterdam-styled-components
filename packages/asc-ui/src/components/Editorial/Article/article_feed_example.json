{
  "type":"node--article",
  "title": " Jeugdwerkloosheid Amsterdam daalt naar 6,2%",
  "image": "http://lorempixel.com/1280/600/sports/1/",
  "publicationDate": "2019-03-15",
  "publicationDateFormatted": "15 maart 2019",
  "body": "\u003Cp\u003EOIS heeft de belangrijkste ontwikkelingen in de jeugdwerkloosheid in beeld gebracht door jongeren op basis van de kenmerken onderwijs, werk en werkloosheid in te delen in zes groepen. De cijfers, gebaseerd op de internationale definitie van werkloosheid, zijn nu voor 2014 tot en met 2018 beschikbaar.\u003C\/p\u003E\n\n\u003Ch4\u003E\u003Cimg alt=\u0022Jeugdwerkloosheid in 2014 en 2018\u0022 data-entity-type=\u0022file\u0022 data-entity-uuid=\u002248be11aa-dc99-4554-b4c0-a2354e390119\u0022 src=\u0022https://acc.cms.data.amsterdam.nl\/sites\/default\/files\/inline-images\/2019_nieuws_jeugdwerkloosheid_1.png\u0022 \/\u003E\u003C\/h4\u003E\n\n\u003Ch4\u003E\u003Cstrong\u003EJeugdwerkloosheid Amsterdam lager dan Rotterdam en Den Haag\u003C\/strong\u003E\u003C\/h4\u003E\n\n\u003Cp\u003EDe totale Amsterdamse jeugdwerkloosheid ligt met 6,2% rond het landelijke gemiddelde (6,5%). In Den Haag (9%), Rotterdam (10%) is nog steeds sprake van hogere jeugdwerkloosheid dan in Amsterdam. In Utrecht (6%) ligt de jeugdwerkloosheid op eenzelfde niveau als in Amsterdam.\u003C\/p\u003E\n\n\u003Cp\u003E\u003Cimg alt=\u0022Werkloze beroepsbevolking 15-26 jaar in Nederland en de grote steden, 2014 en 2018 in procenten\u0022 data-entity-type=\u0022file\u0022 data-entity-uuid=\u0022c94fed90-ff90-42d1-9aec-94985ee92940\u0022 src=\u0022https://acc.cms.data.amsterdam.nl\/sites\/default\/files\/inline-images\/2019_nieuws_jeugdwerkloosheid_2.png\u0022 \/\u003E\u003C\/p\u003E\n\n\u003Ch4\u003E\u003Cstrong\u003EJongere zonder startkwalificatie die geen onderwijs meer volgen vaakst werkloos\u003C\/strong\u003E\u003C\/h4\u003E\n\n\u003Cp\u003ELaagopgeleide jongeren zijn vaker dan andere groepen werkloos (11%), dit geldt het sterkst voor de laag opgeleide jongeren (jongeren zonder startkwalificatie) die geen onderwijs meer volgen (16%). Ook jongeren met een niet-westerse migratieachtergrond (8% werkloos) zijn nog steeds vaker werkloos dan jongeren zonder migratieachtergrond (5% werkloos), maar het verschil is wel kleiner geworden in de afgelopen jaren. Wanneer opleidingsniveau en migratieachtergrond gecombineerd worden, blijkt dat het verschil in werkloosheid tussen jongeren met een niet-westerse migratieachtergrond en jongeren met een Nederlandse achtergrond blijft bestaan voor laag- en middelbaar opgeleide jongeren. Bij hoogopgeleide jongeren is er geen verschil meer in werkloosheid tussen jongeren met een niet-westerse en jongeren met een Nederlandse migratieachtergrond.\u003C\/p\u003E\n\n\u003Cp\u003E\u003Cimg alt=\u0022Ontwikkeling werkloze beroepsbevolking (15-26 jaar) naar opleidingsniveau en migratieachtergrond, 2014 en 2018 in procenten\u0022 data-entity-type=\u0022file\u0022 data-entity-uuid=\u0022aeb811a1-839f-40d6-a9bc-9440b97ebf89\u0022 src=\u0022https://acc.cms.data.amsterdam.nl\/sites\/default\/files\/inline-images\/2019_nieuws_jeugdwerkloosheid_3.png\u0022 \/\u003E\u003C\/p\u003E\n\n\u003Ch4\u003E\u003Cstrong\u003EAantal jongeren met bijstand daalt\u003C\/strong\u003E\u003C\/h4\u003E\n\n\u003Cp\u003EOp 1 januari 2019 ontvingen 2.854 jongeren in de leeftijd van 18 tot en met 26 jaar een uitkering in het kader van de participatiewet (1,9% van de jongeren). In januari 2015 ontvingen 2.769 jongeren een uitkering (2%), 83 minder dan begin 2019. Dat het totaal aantal bijstandsuitkeringen van 18-26 jarigen vrijwel gelijk is gebleven in de afgelopen jaren komt door de instroom van statushouders. Zij zijn, zeker in de eerste jaren na hun komst naar Nederland, nog vaak afhankelijk van bijstand. Als alleen wordt gekeken naar de ontwikkelingen van het aantal jongeren met bijstand exclusief de statushouders dan was er tussen januari 2015 (2.558, 1,8%) en januari 2019 (1.743, 1,2%) sprake van een daling van 33%.\u003C\/p\u003E\n\n\u003Cp\u003E\u003Cimg alt=\u0022Jongeren met bijstand (18-26 jaar), statushouders en niet-statushouders, jan 2015-jan 2019 (aantal)\u0022 data-entity-type=\u0022file\u0022 data-entity-uuid=\u0022a844f04d-ca84-4ae2-aedb-4f77086c586d\u0022 src=\u0022https://acc.cms.data.amsterdam.nl\/sites\/default\/files\/inline-images\/2019_nieuws_jeugdwerkloosheid_4.png\u0022 \/\u003E\u003C\/p\u003E\n\n\u003Ch4\u003E\u003Cstrong\u003EGroot deel bemiddelingen naar school en werk duurzaam\u003C\/strong\u003E\u003C\/h4\u003E\n\n\u003Cp\u003EIn 2015 is de aanpak jeugdwerkloosheid gestart met als doel in vier jaar tijd tenminste 21.000 jongeren naar school of werk begeleiden. Dit is gelukt, er zijn 24.684 jongeren bemiddeld naar een opleiding (18%) of een baan (82%). Van deze bemiddelingen is nagegaan in hoeverre zij duurzaam zijn. Dat wil zeggen minimaal zes maanden onafgebroken onderwijs volgen of werken. Van de jongeren die een opleiding zijn gaan volgen is zes maanden later 83% nog steeds bezig met een opleiding. Van de jongeren die zijn gaan werken is zes maanden later 60% nog steeds werkzaam. Verder blijkt dat bemiddelingen naar een baan voor minimaal 12 uur per week en banen met een vast karakter (bepaalde of onbepaalde tijd) vaker duurzaam zijn dan bemiddelingen naar bijvoorbeeld uitzendbanen. Zie de rapportage bemiddelingen jeugdwerkloosheid 2015-2018 voor meer informatie.\u003C\/p\u003E\n\n\u003Cp\u003E\u003Cimg alt=\u0022Duurzame bemiddeling jeugdwerkloosheid, naar school en werk 2015-2018\u0022 data-entity-type=\u0022file\u0022 data-entity-uuid=\u002242efa13d-3a20-4f77-8e08-ce86af54e92a\u0022 src=\u0022https://acc.cms.data.amsterdam.nl\/sites\/default\/files\/inline-images\/2019_nieuws_jeugdwerkloosheid_5.png\u0022 \/\u003E\u003C\/p\u003E",
  "fields": {
    "intro": "Van de 146.500 Amsterdamse jongeren in de leeftijd van 15 tot en met 26 jaar waren er in 2018 gemiddeld 6.100 werkloos, dit komt neer op 6,2% van de beroepsbevolking. In 2014 was dit nog 13,3% (13.300 jongeren). De Amsterdamse jeugdwerkloosheid lag in 2014 ruim boven het landelijke gemiddeld, inmiddels zijn Amsterdamse jongeren even vaak als gemiddeld in Nederland werkloos.",
    "downloads": {

    }
  }
}
