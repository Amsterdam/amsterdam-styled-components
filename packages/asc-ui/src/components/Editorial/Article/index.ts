export { default as ArticleStyle } from './ArticleStyle'
export { default as Article } from './Article'
