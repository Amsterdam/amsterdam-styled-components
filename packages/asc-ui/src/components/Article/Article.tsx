import React from 'react'
import ArticleStyle, { Props } from './ArticleStyle'
import BlogPost from '../Blog/BlogPost/BlogPost'

const Article: React.FC<Props> = ({ children, image }) => (
  <BlogPost StyledComponent={ArticleStyle} image={image}>
    {children}
  </BlogPost>
)

export default Article
