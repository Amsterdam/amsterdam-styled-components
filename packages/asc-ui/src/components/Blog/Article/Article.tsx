import React from 'react'
import ArticleStyle from './ArticleStyle'
import { Props } from '../BlogPost/BlogPost'

const Article: React.FC<Props & React.HTMLAttributes<HTMLElement>> = ({
  children,
  image,
  className,
}) => (
  <ArticleStyle image={image} className={className}>
    {children}
  </ArticleStyle>
)

export default Article
