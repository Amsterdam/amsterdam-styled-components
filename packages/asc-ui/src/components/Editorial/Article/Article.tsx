import React, { FunctionComponent } from 'react'
import ArticleStyle from './ArticleStyle'
import { Props } from '../EditorialPost/EditorialPost'

const Article: FunctionComponent<Props & React.HTMLAttributes<HTMLElement>> = ({
  children,
  image,
  ...otherProps
}) => (
  <ArticleStyle image={image} {...otherProps}>
    {children}
  </ArticleStyle>
)

export default Article
