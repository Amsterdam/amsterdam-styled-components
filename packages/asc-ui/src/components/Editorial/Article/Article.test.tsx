import { render } from '@testing-library/react'
import Article from './Article'

describe('Article', () => {
  it('should render', () => {
    const { container } = render(
      <Article>This is the publication content</Article>,
    )
    expect(container.firstChild).toBeDefined()
  })
})
