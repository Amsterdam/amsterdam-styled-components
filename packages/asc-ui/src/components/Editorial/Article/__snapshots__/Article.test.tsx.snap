// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article should render 1`] = `
.c1 {
  width: 100%;
  color: #323232;
  background-color: #ffffff;
}

.c1 .c2 {
  max-height: 560px;
  min-height: 250px;
}

@media screen and (min-width:768px) {
  .c0 .sc-bdVaJa {
    margin-top: 0;
  }
}

<section
  class="c0 c1"
>
  This is the publication content
</section>
`;
